PGIP/GraphQL/Resolver/Signature.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module PGIP.GraphQL.Resolver.Signature (resolve) where
+
+import PGIP.GraphQL.Resolver.ToResult
+
+import PGIP.GraphQL.Result as GraphQLResult
+
+import PGIP.Shared
+
+import Driver.Options
+import Persistence.Database
+import Persistence.Utils
+import Persistence.Schema as DatabaseSchema
+
+import Database.Esqueleto
+
+import Control.Monad.IO.Class (MonadIO (..))
+
+resolve :: HetcatsOpts -> Cache -> Int -> IO (Maybe GraphQLResult.Result)
+resolve opts sessionReference idVar =
+  onDatabase (databaseConfig opts) $ resolveDB idVar
+
+resolveDB :: MonadIO m => Int -> DBMonad m (Maybe GraphQLResult.Result)
+resolveDB idVar = do
+  signatureL <- select $ from $ \signatures -> do
+    where_ (signatures ^. SignatureId ==. val (toSqlKey $ fromIntegral idVar))
+    return signatures
+  case signatureL of
+    [] -> return Nothing
+    signatureEntity@(Entity signatureKey _) : _ -> do
+      omsL <- select $ from $ \(omsSql `InnerJoin` loc_id_bases) -> do
+        on (coerceId (omsSql ^. OMSId) ==. loc_id_bases ^. LocIdBaseId)
+        where_ (omsSql ^. OMSSignatureId ==. val signatureKey)
+        return loc_id_bases
+
+      signatureMorphismsAsSourceL <- select $ from $ \signature_morphisms -> do
+        where_ (signature_morphisms ^. SignatureMorphismSourceId ==. val signatureKey)
+        return signature_morphisms
+
```

Comment:
why are signature morphisms are needed here?

Reply:
This way, it's possible to find all translations from/to a signature. At least, this is what is implemented in the backend.